dagmc/dagnav.py

```diff
@@ -224,6 +224,40 @@ def _get_triangle_sets(self):
 
 class Volume(DAGSet):
 
+    @property
+    def groups(self) -> list[Group]:
+        """Get list of groups containing this volume."""
+        return [group for group in self.model.groups.values() if self in group]
+
+    @property
+    def material(self) -> Optional[str]:
+        """Name of the material assigned to this volume."""
+        for group in self.groups:
+            if self in group and group.name.startswith("mat:"):
+                return group.name[4:]
+        return None
+
+    @material.setter
+    def material(self, name: str):
+        existing_group = False
+        for group in self.model.groups.values():
+            if f"mat:{name}" == group.name:
+                # Add volume to group matching specified name, unless the volume
+                # is already in it
+                if self in group:
+                    return
+                group.add_set(self)
+                existing_group = True
+
+            elif self in group and group.name.startswith("mat:"):
+                # Remove volume from existing group
+                group.remove_set(self)
+
+        if not existing_group:
+            # Create new group, add name/category tags, add entity
+            new_group = Group.create(self.model, name=f"mat:{name}")
```

Comment:
We should probably provide a new Group ID here as well. They don't technically have an impact but it would make the new set complete.